### PR TITLE
chore: add support for python 3.10 in smclarify

### PR DESCRIPTION
**Pull request by Satish615**
*Description of changes:*
1. update smclairfy to ```0.4``` to support api changes the bias_stat metrics and model performance report.
2. update github workflows to build the packages on python ```3.7, 3.8, 3.9, 3.10 ```and run the test-coverage matrix for python ```3.9 and 3.10```
3.  update the read.me, setup.py , setup.cfg to support ```python 3.10```.
4. fix ValueError with sns.countplot() in example notebooks. 
```
ValueError: The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
```

*Testing:*
./devtool all passed in local.
github actions workflow passed for 3.7, 3.8 and 3.9, 3.10 on the [forked repo](https://github.com/Satish615/amazon-sagemaker-clarify/actions/runs/4410316162/). 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8, 3.7, 3.9]
+        python-version: ['3.8', '3.7', '3.9', '3.10']
```
Comment (goswamig) [inline .github/workflows/python-package.yml:13]: Just curious, did we first support 3.8 and then 3.7 ?

Comment (Satish615) [reply .github/workflows/python-package.yml:13]: no, It was 3.7 first, some how the order was not serail in test matrix.

**Comment by goswamig** [comment]
BTW I noticed we don't have coverage for all bias metrics in our test here we have total 21 bias metrics (pre and post train). 

but report reflect 19 or 20 https://github.com/Satish615/amazon-sagemaker-clarify/actions/runs/4410316162/jobs/7727661324 ?

**Review by spoorn — APPROVED** [review]

File: src/smclarify/__init__.py [hunk 2]
```diff
@@ -1 +1 @@
-__version__ = "0.3"
+__version__ = "0.4"
```
Comment (spoorn) [inline src/smclarify/__init__.py:1]: Do we actually need to bump up the lib version?

Comment (Satish615) [reply src/smclarify/__init__.py:1]: version bump up is not for the python 3.10 support. we made some api changes for bias_stats metrics and model report.

**Comment by Satish615** [comment]
> BTW I noticed we don't have coverage for all bias metrics in our test here we have total 21 bias metrics (pre and post train).
> 
> but report reflect 19 or 20 https://github.com/Satish615/amazon-sagemaker-clarify/actions/runs/4410316162/jobs/7727661324 ?

Nice catch! I have verfied that we don't have unit test coverage for 3 metrics TVD(pre), CDDPL(post), DPPL (post). I am adding a tracking sim for these.

**Review by goswamig — APPROVED** [review]

**Pull request merged**